compiler/src/dmd/root/longdouble.d

```diff
@@ -229,14 +220,20 @@ else version(D_InlineAsm_X86_64)
     private:
     string fld_arg(string arg)()
     {
-        return "asm nothrow @nogc pure @trusted { mov RAX, " ~ arg ~ "; fld real ptr [RAX]; }";
+        return "asm nothrow @nogc pure @trusted { lea RAX, " ~ arg ~ "; fld real ptr [RAX]; }";
     }
     string fstp_arg(string arg)()
+    {
+        return "asm nothrow @nogc pure @trusted { lea RAX, " ~ arg ~ "; fstp real ptr [RAX]; }";
+    }
+    string fld_parg(string arg)()
+    {
+        return "asm nothrow @nogc pure @trusted { mov RAX, " ~ arg ~ "; fld real ptr [RAX]; }";
+    }
+    string fstp_parg(string arg)()
     {
         return "asm nothrow @nogc pure @trusted { mov RAX, " ~ arg ~ "; fstp real ptr [RAX]; }";
     }
-    alias fld_parg = fld_arg;
-    alias fstp_parg = fstp_arg;
```

Comment:
You likely are going to have to split this between MSVC and Posix.

https://d.godbolt.org/z/WzG4qzKT7

Reply:
Yep

Reply:
Is there a gdc friendly equivalent of the version(LDC) code? Don't really want so much inline asm if possible

Reply:
I think this is equivalent, LDC also accepts it and compiles to the same code.

https://d.godbolt.org/z/aee3asT15